### PR TITLE
Namespaces and naming

### PR DESCRIPTION
**Pull request by ondrejbouda**
#11

### CODE REVIEW

File: src/MessageBus/MessageBus.php [hunk 1]
```diff
@@ -22,24 +22,22 @@ final class MessageBus implements IMessageBus
...
-	}
-
-
-
-	public function appendMiddleware(IMessageBusMiddleware $middleware) : void
-	{
-		$this->middleware[] = $middleware;
 
-		$this->cachedCallback = NULL;
+		$this->cachedCallback = $this->createMiddlewareCallback();
```
Comment (xificurk) [inline src/MessageBus/MessageBus.php]: We doing work in constructor are? Yrsssss. :thinking:

File: src/MessageBus/MessageBus.php [hunk 2]
```diff
@@ -22,24 +22,22 @@ final class MessageBus implements IMessageBus
...
+	private $cachedCallback;
 
 
 
-	public function __construct(?MiddlewareCallbackChainCreator $middlewareCallbackChainCreator = NULL)
+	/**
+	 * @param IMessageBusMiddleware[] $middleware
+	 * @param MiddlewareCallbackChainCreator|null $middlewareCallbackChainCreator
+	 */
+	public function __construct(array $middleware, ?MiddlewareCallbackChainCreator $middlewareCallbackChainCreator = NULL)
```
Comment (xificurk) [inline src/MessageBus/MessageBus.php]: Mabe consider use of variadic to add more type-safety `IMessageBusMiddleware ...$middleware`

Comment (ondrejbouda) [reply src/MessageBus/MessageBus.php]: Will do.

File: tests/Integration/CommandHandlingTest.phpt [hunk 3]
```diff
@@ -51,11 +51,13 @@ class CommandHandlingTest extends DjTestCase
 
 		$handlerInvoker = new HandlerInvoker();
 
-		$bus = new MessageBus([
-			new HandlerTypesResolvingMiddleware($handlerTypesResolver),
-			new SplitByHandlerTypeMiddleware(),
-			new HandlerInvokingMiddleware($handlerProvider, $handlerInvoker),
-		]);
+		$bus = new MessageBus(
+			...[
```
Comment (xificurk) [inline tests/Integration/CommandHandlingTest.phpt]: :smiley: :question:

**Pull request merged**